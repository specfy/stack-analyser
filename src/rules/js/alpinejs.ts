import { register } from '../../register.js';

register({
  tech: 'alpinejs',
  name: 'Alpine.js',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'alpinejs' }],
});
