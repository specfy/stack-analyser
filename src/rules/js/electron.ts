import { register } from '../../register.js';

register({
  tech: 'electron',
  name: 'Electron',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'electron' }],
});
