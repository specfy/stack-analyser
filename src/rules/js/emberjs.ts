import { register } from '../../register.js';

register({
  tech: 'emberjs',
  name: 'Ember',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'ember-cli' },
    { type: 'npm', name: 'ember' },
    { type: 'npm', name: 'ember-data' },
    { type: 'npm', name: 'ember-source' },
  ],
});
