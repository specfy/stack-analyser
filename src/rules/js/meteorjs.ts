import { register } from '../../register.js';

register({
  tech: 'meteorjs',
  name: 'Meteor',
  type: 'tool',
  files: ['.meteor'],
  dependencies: [
    { type: 'npm', name: 'meteor' },
    { type: 'npm', name: 'meteor-node-stubs' },
  ],
});
