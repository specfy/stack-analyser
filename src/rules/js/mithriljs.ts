import { register } from '../../register.js';

register({
  tech: 'mithriljs',
  name: 'Mithril',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'mithril' }],
});
