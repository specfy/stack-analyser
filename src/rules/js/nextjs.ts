import { register } from '../../register.js';

register({
  tech: 'nextjs',
  name: 'Next.js',
  type: 'tool',
  files: ['next.config.js'],
  dependencies: [{ type: 'npm', name: 'next' }],
});
