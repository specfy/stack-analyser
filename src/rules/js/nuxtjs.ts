import { register } from '../../register.js';

register({
  tech: 'nuxtjs',
  name: 'Nuxt.js',
  type: 'tool',
  files: ['nuxt.config.js'],
  dependencies: [{ type: 'npm', name: 'nuxt' }],
});
