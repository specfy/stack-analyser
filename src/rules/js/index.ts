import './alpinejs.js';
import './angular.js';
import './bootstrap.js';
import './emberjs.js';
import './esbuild.js';
import './eslint.js';
import './express.js';
import './fastify.js';
import './koa.js';
import './meteorjs.js';
import './mithriljs.js';
import './nextjs.js';
import './nuxtjs.js';
import './preactjs.js';
import './prettier.js';
import './prisma.js';
import './puppeteer.js';
import './react.js';
import './reactEmail.js';
import './rollup.js';
import './sequelize.js';
import './socketio.js';
import './solidjs.js';
import './storybook.js';
import './strapi.js';
import './stylelint.js';
import './sveltejs.js';
import './tailwind.js';
import './typescript.js';
import './vite.js';
import './vue.js';
import './webpack.js';
