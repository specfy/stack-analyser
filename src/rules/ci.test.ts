import { describe, expect, it } from 'vitest';

import { analyser } from '../analyser/index.js';
import type { AllowedKeys } from '../index.js';
import { rawList } from '../loader.js';
import { FakeProvider } from '../provider/fake.js';
import { listIndexed } from '../register.js';

import '../autoload.js';

describe('ci', () => {
  it('should match everything', async () => {
    const paths: string[] = [];
    for (const item of rawList) {
      if (
        item.type !== 'file' ||
        listIndexed[item.ref.tech].type !== 'ci' ||
        !item.ref.files
      ) {
        continue;
      }

      paths.push('example' in item.ref ? item.ref.example : item.ref.files[0]);
    }

    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': paths,
        },
        files: {
          '/.github/workflows/ci.yml': '', // we open this file
        },
      }),
    });
    expect(res.toJson('').techs).toMatchSnapshot();
  });

  it('enforce that we match .github', async () => {
    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['.github/'],
          '/.github': ['workflows/'],
          '/.github/workflows': ['main.yml'],
        },
        files: {
          '/.github/workflows/main.yml': '',
        },
      }),
    });
    const match: AllowedKeys[] = ['github', 'github.actions'];
    expect(res.toJson('').techs).toStrictEqual(match);
  });
});
