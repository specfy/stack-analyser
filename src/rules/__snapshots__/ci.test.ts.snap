// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ci > should match everything 1`] = `
[
  "appveyor",
  "azure.ci",
  "browserstack",
  "circleci",
  "cirrusci",
  "codesandboxci",
  "cypressci",
  "dependabot",
  "github.actions",
  "gitlab.ci",
  "jenkins",
  "relativeci",
  "renovate",
  "styleci",
  "teamcity",
  "travisci",
  "trufflesecurity",
]
`;
